### ISSUE TITLE
Code about FC_model with XE loss

### ISSUE THREAD

**Issue by miracle24** [ruotianluo/self-critical.pytorch#15]
I noticed that according to you code about the FC model parts, the FC model will generate a sequence without SOS (start of sentence) but with EOS (end of sentence), 
`return torch.cat([_.unsqueeze(1) for _ in outputs[1:]], 1).contiguous() #from forward() in FCModel.py`
but, when the XE loss is calculated in LanguageModelCriterion, the mask will not cover the EOS: 
`mask =  mask[:, :input.size(1)]`. 
So in fact, in your code, the XE loss is just calculated without these two special token. Is it right to calculate the XE loss in this way. Why do you leave out the SOS and EOS?

**Comment by ruotianluo** [comment]
I don't leave out eos, you can check. Start token is left out because you don't generate start token.

**Comment by miracle24** [comment]
My fault. The eos is not left out.

**Comment by miracle24** [comment]
Start token is input into the LSTM at the first time step (w_0 = image feature), although it is not considered when calculating the score. So I think the loss of 'sos' could be taken into account.

**Comment by ruotianluo** [comment]
But it's always the input at first time step. you don't need to predict it so you don't need to add the loss for it.

**Comment by miracle24** [comment]
It occurs to me that in function sample() in FCModel.py, the input of LSTM at t=1 is the start token, not one sampled from the model distribution at t=0.  So I am wrong. Thanks.
